### ISSUE TITLE
Hard Coded Formatting Parameters

### ISSUE THREAD

**Issue by flipdazed** [flipdazed/SoftwareDevelopment#97]
**Bugged Version** b8ebc59170182cc817852a2ee5c62c158a2d8310
**File** `logger_formater.py`
**Lines** `6-33`

**Issue** 

To document risk the following code should be adjusted as values are hardcoded in the class.
There is also a strange function being declared from global variables. A change was attempted in #89 to make this more dynamic but failed. Would require investigation.

```
### Create new Logging level for game messages ###
# set up custom logging level or game output
GAME = 25
logging.addLevelName(GAME, "GAME")
logging.GAME = GAME
def game(self, message, *args, **kwargs):
    # Yes, logger takes its '*args' as 'args'
    if self.isEnabledFor(GAME):
        self._log(GAME, message, args, **kwargs)
logging.Logger.game = game
### End creation of new logging level ####

# Custom formatter for logging
# http://stackoverflow.com/a/8349076/4013571
class MyFormatter(logging.Formatter):
    """Class to create custom formats"""

    # these lines seem a bit awkward but I don't know 
    # enought python to change them!
    err_fmt  = "ERROR: %(module)s: %(lineno)d: %(msg)s"
    dbg_fmt  = "DEBUG: %(module)s: %(lineno)d: %(msg)s"
    info_fmt = "INFO: %(module)s: %(msg)s"
    game_fmt = "GAME: %(msg)s"
```

**Fix**
Investigate inner functionality of `logging` module and create dynamic display settings

**Comment by flipdazed** [comment]
set to won't fix and assigned to future development #95

**Comment by flipdazed** [comment]
had to reopen to get colours to work

more pain in python copying objects!!!

```
def unit_test():
    """for some reason this won't work"""

    def copy_func(f, name=None):
        return types.FunctionType(f.func_code, f.func_globals, name or f.func_name,
            f.func_defaults, f.func_closure)

    ### Create new Logging level for game messages ###
    # set up custom logging level or game output
    LOGGER_NAMES = {"GAME":25, "USER":26, "COMPUTER":27}

    for key, val in LOGGER_NAMES.iteritems():

        nkey = deepcopy(key)
        nval = deepcopy(val)

        # adds the logging level
        logging.addLevelName(nkey, oval)
        # sets the attribute
        setattr(logging, nkey, oval)
        # adds the formatter
        def fmt_fn(self, message, *args, **kwargs):
            # Yes, logger takes its '*args' as 'args'
            if self.isEnabledFor(oval):
                self._log(nval, message, args, **kwargs)

        setattr(logging.Logger, nkey.lower(), copy_func(fmt_fn, key))

```

Solution is to hardcode the functions as:

```
def create_new_levels():
    ### Create new Logging level for game messages ###
    # set up custom logging level or game output
    GAME = 25
    logging.addLevelName(GAME, "GAME")
    logging.GAME = GAME
    def game(self, message, *args, **kwargs):
        # Yes, logger takes its '*args' as 'args'
        if self.isEnabledFor(GAME):
            self._log(GAME, message, args, **kwargs)
    logging.Logger.game = game
    ### End creation of new logging level ####

    ### Create new Logging level for game messages ###
    # set up custom logging level or game output
    USER = 26
    logging.addLevelName(USER, "USER")
    logging.USER = USER
    def user(self, message, *args, **kwargs):
        # Yes, logger takes its '*args' as 'args'
        if self.isEnabledFor(USER):
            self._log(USER, message, args, **kwargs)
    logging.Logger.user = user
    ### End creation of new logging level ####

    ### Create new Logging level for USER messages ###
    # set up custom logging level or computer output
    COMPUTER = 27
    logging.addLevelName(COMPUTER, "COMPUTER")
    logging.COMPUTER = COMPUTER
    def computer(self, message, *args, **kwargs):
        # Yes, logger takes its '*args' as 'args'
        if self.isEnabledFor(COMPUTER):
            self._log(COMPUTER, message, args, **kwargs)
    logging.Logger.computer = computer
    ## End creation of new logging level ####
    pass

```
